internal/test/test_tpm.go

```diff
@@ -40,6 +41,26 @@ func (s simulatedTpm) EventLog() ([]byte, error) {
 	return s.eventLog, nil
 }
 
+// SkipOnAlgorithm skips the test if the algorithm is not found in the TPM
+// capability.
+func SkipOnAlgorithm(tb testing.TB, tpm io.ReadWriteCloser, alg tpm2.Algorithm) {
+	tb.Helper()
+
+	caps, moreData, err := tpm2.GetCapability(tpm, tpm2.CapabilityAlgs, math.MaxUint32, 0)
```

Comment:
While I suspect `math.MaxUint32` as propertyCount works in theory, the spec says the following (part 3, 30.2.1):
```
When all of the properties of the requested type have been returned, the moreData parameter in the
response will be set to NO. Otherwise, it will be set to YES.
NOTE 3 The moreData parameter will be YES if there are more properties even if the requested number of
capabilities has been returned.
The TPM is not required to return more than one value at a time. It is not required to provide the same
number of values in response to subsequent requests.
EXAMPLE 2 A TPM may return 4 properties in response to a TPM2_GetCapability(capability =
TPM_CAP_TPM_PROPERTY, property = TPM_PT_MANUFACTURER, propertyCount = 8 ) and for a
latter request with the same parameters, the TPM may return as few as one and as many as 8
values.
```

So, a TPM may in fact only return 1 property with moreData set to true and we would get the Fatal `received extra data from GetCapability`

Unless we have a good reason for not, we should use the moreData field to collect all properties.

Reply:
Yes, I will use the implementation in GoTPM's skipOnUnsupportedAlgorithm which handles moreData.